src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/AuroraTopologyService.java

```diff
@@ -190,30 +180,25 @@ public void setClusterInstanceTemplate(HostInfo clusterInstanceTemplate) {
   @Override
   public List<HostInfo> getTopology(JdbcConnection conn, boolean forceUpdate)
       throws SQLException {
-    ClusterTopologyInfo clusterTopologyInfo = topologyCache.get(this.clusterId);
 
-    if (clusterTopologyInfo == null
-        || Util.isNullOrEmpty(clusterTopologyInfo.hosts)
-        || forceUpdate
-        || refreshNeeded(clusterTopologyInfo)) {
+    List<HostInfo> hosts = topologyCache.get(this.clusterId, this.refreshRateNanos);
```

Comment:
It looks like the CacheMap `get` implementation will keep on updating the expiration time of the entry each time we get here. This means if we keep fetching the cached item before the expiry, we will keep on updating the expiration time and returning the cached item. Is this what we want? I would've expected we set the expiry time once only, when the entry is first created. That way we would avoid returning a cached item that could be very old if we've been fetching it from the cache regularly